lib/rubocop/cop/style/redundant_regexp_character_class.rb

```diff
@@ -65,19 +48,54 @@ def on_regexp(node)
         private
 
         def each_redundant_character_class(node)
-          pattern_source(node).scan(PATTERN) do
-            yield match_range(node.loc.begin.end, Regexp.last_match)
+          each_single_element_character_class(node) do |char_class|
+            next unless redundant_single_element_character_class?(node, char_class)
+
+            yield node.loc.begin.adjust(begin_pos: 1 + char_class.ts, end_pos: char_class.te)
+          end
+        end
+
+        def each_single_element_character_class(node)
+          Regexp::Parser.parse(pattern_source(node)).each_expression do |expr|
+            next if expr.type != :set || expr.expressions.size != 1
+            next if expr.negative? || %i[posixclass set].include?(expr.expressions.first.type)
+
+            yield expr
           end
+        rescue Regexp::Scanner::ScannerError
```

Comment:
This rescue is no longer needed, the gem was fixed.

Reply:
Hmm, maybe I'm missing something - but if I don't add a rescue here, then I get a failure in this example:
```
  context "with a regexp containing invalid \g escape" do
    it 'does not register an offense' do
      # See https://ruby-doc.org/core-2.7.1/Regexp.html#class-Regexp-label-Subexpression+Calls
      # \g should be \g<name>
      expect_no_offenses('foo = /[a]\g/')
    end
  end
```
```
Failures:

  1) RuboCop::Cop::Style::RedundantRegexpCharacterClass with a regexp containing invalid g escape does not register an offense
     Failure/Error:
                 Regexp::Parser.parse(pattern_source(node)).each_expression do |expr|
                   next if expr.type != :set || expr.expressions.size != 1
                   next if expr.negative? || %i[posixclass set].include?(expr.expressions.first.type)

                   yield expr
                 end

     Regexp::Scanner::ScannerError:
       Scan error at '\g'
     # ./lib/rubocop/cop/style/redundant_regexp_character_class.rb:57:in `each_single_element_character_class'
     # ./lib/rubocop/cop/style/redundant_regexp_character_class.rb:49:in `each_redundant_character_class'
     # ./lib/rubocop/cop/style/redundant_regexp_character_class.rb:33:in `on_regexp'
     # ./lib/rubocop/cop/commissioner.rb:99:in `block (2 levels) in trigger_responding_cops'
     # ./lib/rubocop/cop/commissioner.rb:152:in `with_cop_error_handling'
     # ./lib/rubocop/cop/commissioner.rb:98:in `block in trigger_responding_cops'
     # ./lib/rubocop/cop/commissioner.rb:97:in `each'
     # ./lib/rubocop/cop/commissioner.rb:97:in `trigger_responding_cops'
     # ./lib/rubocop/cop/commissioner.rb:70:in `on_regexp'
     # /Users/owen/code/rubocop-ast/lib/rubocop/ast/traversal.rb:70:in `on_lvasgn'
     # ./lib/rubocop/cop/commissioner.rb:72:in `on_lvasgn'
     # /Users/owen/code/rubocop-ast/lib/rubocop/ast/traversal.rb:14:in `walk'
     # ./lib/rubocop/cop/commissioner.rb:85:in `investigate'
     # ./lib/rubocop/cop/team.rb:152:in `investigate_partial'
     # ./lib/rubocop/cop/team.rb:83:in `investigate'
     # ./lib/rubocop/rspec/cop_helper.rb:54:in `_investigate'
     # ./lib/rubocop/rspec/cop_helper.rb:25:in `inspect_source'
     # ./lib/rubocop/rspec/expect_offense.rb:187:in `expect_no_offenses'
     # ./spec/rubocop/cop/style/redundant_regexp_character_class_spec.rb:68:in `block (3 levels) in <top (required)>'
```

Reply:
Interesting. I'm opened an issue on the gem about this 👍

Reply:
https://github.com/ammar/regexp_parser/issues/65 for the record 👍